kroxylicious-multitenant/src/main/java/io/kroxylicious/proxy/filter/multitenant/MultiTenantTransformationFilter.java

```diff
@@ -235,6 +255,54 @@ public void onListGroupsResponse(ResponseHeaderData header, ListGroupsResponseDa
         context.forwardResponse(response);
     }
 
+    @Override
+    public void onJoinGroupRequest(RequestHeaderData header, JoinGroupRequestData request, KrpcFilterContext context) {
+        var tenantPrefix = getTenantPrefix(context);
+        request.setGroupId(tenantPrefix + request.groupId());
+        context.forwardRequest(request);
+    }
+
+    @Override
+    public void onJoinGroupResponse(ResponseHeaderData header, JoinGroupResponseData response, KrpcFilterContext context) {
```

Comment:
We should get rid of these otherwise empty methods.  The unit test harness will need a small refactor to allow that to work.

Reply:
I agree with the sentiment, can you shed some light on why (and how) the test framework needs to change?

Reply:
We had to change the test framework so that it's able to filter null requests/responses from the test definitions - it's been pushed as a separate commit now!